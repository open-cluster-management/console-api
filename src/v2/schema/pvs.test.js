/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */

import supertest from 'supertest';
import server, { GRAPHQL_PATH } from '../index';

describe('PVs Resolver', () => {
  test('Correctly Resolves PVs Query', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        {
          pvs {
            accessModes
            capacity
            claim
            claimRef
            cluster {
              metadata {
                name
              }
            }
            metadata{
              name
              namespace
            }
            reclaimPolicy
            status
            type
          }
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));

  test('Correctly Resolves PVsClaims Query', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        {
          pvsClaims {
            accessModes
            cluster {
              metadata {
                name
              }
            }
            metadata {
              name
              namespace
            }
            persistentVolume
            requests
            status
          }
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));
});
