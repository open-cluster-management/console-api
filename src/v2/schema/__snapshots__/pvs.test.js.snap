// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PVs Resolver Correctly Resolves PVs Query 1`] = `
Object {
  "data": Object {
    "pvs": Array [
      Object {
        "accessModes": Array [
          "ReadWriteOnce",
        ],
        "capacity": "5Gi",
        "claim": "/var/lib/icp/helmrepo",
        "claimRef": Object {
          "name": "helm-repo-pvc",
          "namespace": "kube-system",
        },
        "cluster": Object {
          "metadata": Object {
            "name": "mycluster",
          },
        },
        "metadata": Object {
          "name": "helm-repo-pv",
          "namespace": null,
        },
        "reclaimPolicy": "Delete",
        "status": "Bound",
        "type": "Hostpath",
      },
      Object {
        "accessModes": Array [
          "ReadWriteOnce",
        ],
        "capacity": "20Gi",
        "claim": "/var/lib/registry",
        "claimRef": Object {
          "name": "image-manager-image-manager-0",
          "namespace": "kube-system",
        },
        "cluster": Object {
          "metadata": Object {
            "name": "mycluster",
          },
        },
        "metadata": Object {
          "name": "image-manager-9.42.23.230",
          "namespace": null,
        },
        "reclaimPolicy": "Retain",
        "status": "Bound",
        "type": "LocalVolume",
      },
      Object {
        "accessModes": Array [
          "ReadWriteOnce",
        ],
        "capacity": "20Gi",
        "claim": "/var/lib/icp/logging/elk-data",
        "claimRef": Object {
          "name": null,
          "namespace": null,
        },
        "cluster": Object {
          "metadata": Object {
            "name": "mycluster",
          },
        },
        "metadata": Object {
          "name": "logging-datanode-9.42.23.230",
          "namespace": null,
        },
        "reclaimPolicy": "Retain",
        "status": "Available",
        "type": "LocalVolume",
      },
      Object {
        "accessModes": Array [
          "ReadWriteOnce",
        ],
        "capacity": "10Gi",
        "claim": "/var/lib/mysql",
        "claimRef": Object {
          "name": "mysqldata-mariadb-0",
          "namespace": "kube-system",
        },
        "cluster": Object {
          "metadata": Object {
            "name": "mycluster",
          },
        },
        "metadata": Object {
          "name": "mariadb-9.42.23.230",
          "namespace": null,
        },
        "reclaimPolicy": "Retain",
        "status": "Bound",
        "type": "LocalVolume",
      },
      Object {
        "accessModes": Array [
          "ReadWriteOnce",
        ],
        "capacity": "5Gi",
        "claim": "/var/lib/icp/mgmtrepo",
        "claimRef": Object {
          "name": "mgmt-repo-pvc",
          "namespace": "kube-system",
        },
        "cluster": Object {
          "metadata": Object {
            "name": "mycluster",
          },
        },
        "metadata": Object {
          "name": "mgmt-repo-pv",
          "namespace": null,
        },
        "reclaimPolicy": "Delete",
        "status": "Bound",
        "type": "Hostpath",
      },
      Object {
        "accessModes": Array [
          "ReadWriteOnce",
        ],
        "capacity": "20Gi",
        "claim": "/var/lib/icp/mongodb",
        "claimRef": Object {
          "name": "mongodbdir-icp-mongodb-0",
          "namespace": "kube-system",
        },
        "cluster": Object {
          "metadata": Object {
            "name": "mycluster",
          },
        },
        "metadata": Object {
          "name": "mongodb-9.42.23.230",
          "namespace": null,
        },
        "reclaimPolicy": "Retain",
        "status": "Bound",
        "type": "LocalVolume",
      },
    ],
  },
}
`;

exports[`PVs Resolver Correctly Resolves PVsClaims Query 1`] = `
Object {
  "data": Object {
    "pvsClaims": Array [
      Object {
        "accessModes": Array [
          "ReadWriteOnce",
        ],
        "cluster": Object {
          "metadata": Object {
            "name": "mycluster",
          },
        },
        "metadata": Object {
          "name": "helm-repo-pvc",
          "namespace": "kube-system",
        },
        "persistentVolume": "helm-repo-pv",
        "requests": "5Gi",
        "status": "Bound",
      },
      Object {
        "accessModes": Array [
          "ReadWriteOnce",
        ],
        "cluster": Object {
          "metadata": Object {
            "name": "mycluster",
          },
        },
        "metadata": Object {
          "name": "image-manager-image-manager-0",
          "namespace": "kube-system",
        },
        "persistentVolume": "image-manager-9.42.23.230",
        "requests": "20Gi",
        "status": "Bound",
      },
      Object {
        "accessModes": Array [
          "ReadWriteOnce",
        ],
        "cluster": Object {
          "metadata": Object {
            "name": "mycluster",
          },
        },
        "metadata": Object {
          "name": "mgmt-repo-pvc",
          "namespace": "kube-system",
        },
        "persistentVolume": "mgmt-repo-pv",
        "requests": "5Gi",
        "status": "Bound",
      },
      Object {
        "accessModes": Array [
          "ReadWriteOnce",
        ],
        "cluster": Object {
          "metadata": Object {
            "name": "mycluster",
          },
        },
        "metadata": Object {
          "name": "mongodbdir-icp-mongodb-0",
          "namespace": "kube-system",
        },
        "persistentVolume": "mongodb-9.42.23.230",
        "requests": "20Gi",
        "status": "Bound",
      },
      Object {
        "accessModes": Array [
          "ReadWriteOnce",
        ],
        "cluster": Object {
          "metadata": Object {
            "name": "mycluster",
          },
        },
        "metadata": Object {
          "name": "mysqldata-mariadb-0",
          "namespace": "kube-system",
        },
        "persistentVolume": "mariadb-9.42.23.230",
        "requests": "10Gi",
        "status": "Bound",
      },
    ],
  },
}
`;
